cli/command/swarm/init.go

```diff
@@ -44,21 +47,87 @@ func newInitCommand(dockerCli command.Cli) *cobra.Command {
 	flags.BoolVar(&opts.forceNewCluster, "force-new-cluster", false, "Force create a new cluster from current state")
 	flags.BoolVar(&opts.autolock, flagAutolock, false, "Enable manager autolocking (requiring an unlock key to start a stopped manager)")
 	flags.StringVar(&opts.availability, flagAvailability, "active", `Availability of the node ("active"|"pause"|"drain")`)
+	flags.StringVar(&opts.defaultAddrPool, flagDefaultAddrPool, "", "List of default subnet addresses followed by subnet size (format: <subnet,subnet,..:subnet-size)")
+	flags.SetAnnotation(flagDefaultAddrPool, "version", []string{"1.39"})
 	addSwarmFlags(flags, &opts.swarmOptions)
 	return cmd
 }
 
+// getDefaultAddrPool extracts info from address pool string and fills PoolsOpt struct
+func getDefaultAddrPool(defaultAddrPool string) ([]*net.IPNet, int, error) {
+	var (
+		size            int
+		err             error
+		defaultPoolList []*net.IPNet
+	)
+	if defaultAddrPool == "" {
+		// defaultAddrPool is not defined
+		return nil, 0, nil
+	}
+
+	result := strings.Split(defaultAddrPool, ":")
+	if len(result) > 2 {
+		return nil, size, fmt.Errorf("Invalid default address pool format. Expected format CIDR[,CIDR]*:SUBNET-SIZE")
```

Comment:
This will break for IPv6 literals, including IPv4-mapped addresses such as `::ffff:192.0.2.0` which we could support even if actual IPv6 subnets are not supported.

Instead you could do something like this:
```split := strings.LastIndex(defaultAddrPool, ":")
        if split == -1 {
                return nil, size, fmt.Errorf(...)
        }
                ...
        size, err := strconv.Atoi(strings.TrimSpace(defaultAddrPool[split+1:]))
        ...
        subnetlist := strings.Split(defaultAddrPool[:split], ",")
        ...
```

Reply:
I dont think we are going to use IPV6 or IPV4 mapped address format here.

Reply:
As we were discussing a while back, we can use two separate flags; one to set the CIDR, and one to set the subnetmask/size. The flag for CIDR can be specified multiple times, that way we don't even have to do the manual splitting.